### PR TITLE
feat: support authservice with redis, switch to pepr helm chart

### LINKED ISSUE #518: Configure authservice to use Redis
The existing authservice chart uses an in memory session store which makes it impossible to deploy in an HA mode.

### PR DESCRIPTION
**Pull request by rjferguson21**
## Description
* Updates Pepr to 0.34.0
* Uses Pepr Helm Chart
* Adds support to configure redis as authservice memory store

## Related Issue

Fixes https://github.com/defenseunicorns/uds-core/issues/518

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: src/uds-core-config/chart/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: uds-core-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "uds-core-config.labels" . | nindent 4 }}
+type: Opaque
+data:
+{{- range $key, $value := .Values.operator }}
+  {{ $key }}: {{ $value | b64enc | quote }}
+{{- end }}
```
Comment (mjnagel) [inline src/uds-core-config/chart/templates/secret.yaml]: nit: newline

File: src/istio/common/manifests/pepr-istio-config.yaml [hunk 2]
```diff
@@ -21,3 +21,18 @@ spec:
...
+  namespace: pepr-system
+spec:
+  selector:
+    matchLabels:
+      pepr.dev/controller: watcher
+  mtls:
+    mode: STRICT
+  portLevelMtls:
+    "3000":
+      mode: PERMISSIVE
```
Comment (mjnagel) [inline src/istio/common/manifests/pepr-istio-config.yaml:38]: Just for my memory, is this primarily to handle metrics?

**Review by mjnagel — COMMENTED** [review]
LGTM overall - still need to deploy locally to validate metrics/redis setup.

File: src/istio/common/manifests/pepr-istio-config.yaml (hunk 2 above)
Comment (rjferguson21) [reply src/istio/common/manifests/pepr-istio-config.yaml:38]: Yes

File: packages/slim-dev/zarf.yaml [hunk 3]
```diff
@@ -46,6 +46,27 @@ components:
...
       path: ../../dist
       name: module
+    charts:
+      - name: module
+        valuesFiles:
+          - ../../src/pepr/values.yaml
+    actions:
+      onDeploy:
+        before:
+          - cmd: ./zarf tools kubectl annotate secret -n pepr-system pepr-uds-core-api-token meta.helm.sh/release-name=module --overwrite || true
```
Comment (mjnagel) [inline packages/slim-dev/zarf.yaml]: As discussed synchronously, I think we can pull this into a single `src/pepr/zarf.yaml`, potentially along with the `uds-operator-config` chart. Then we can simplify this import a bit and de-dupe the action portion.

File: src/pepr/zarf.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
+kind: ZarfPackageConfig
+metadata:
+  name: pepr-uds-core
+  description: 'Pepr Module: A collection of capabilities for UDS Core'
+  url: https://github.com/defenseunicorns/pepr
+  version: 0.5.0
```
Comment (mjnagel) [inline src/pepr/zarf.yaml]:
```suggestion
```

I think we can/should drop this version for simplifying the create/deploy lifecycle.

File: tasks/deploy.yaml [hunk 5]
```diff
@@ -31,7 +31,7 @@ tasks:
       - description: "Deploy the Pepr Module"
         cmd: |
           PEPR_VERSION=$(npm pkg get version | tr -d '"')
-          uds zarf package deploy build/zarf-package-pepr-uds-core-${UDS_ARCH}-${PEPR_VERSION}.tar.zst --confirm --no-progress --set UDS_SINGLE_TEST=true
+          uds zarf package deploy build/zarf-package-pepr-uds-core-${UDS_ARCH}-${PEPR_VERSION}.tar.zst --confirm --no-progress --set UDS_SINGLE_TEST=true --set PEPR_SERVICE_MONITORS=false
```
Comment (mjnagel) [inline tasks/deploy.yaml]: If you go with the above suggestions we should be able to drop `PEPR_VERSION` from here.

**Review by mjnagel — COMMENTED** [review]
LGTM otherwise!

File: src/pepr/zarf.yaml (hunk 4 above)
Comment (rjferguson21) [reply src/pepr/zarf.yaml]: Updated

**Review by mjnagel — APPROVED** [review]

**Review by mjnagel — CHANGES_REQUESTED** [review]
Two quick asks:
- PR title should probably indicate more than just the pepr update, maybe `feat: support authservice with redis, switch to pepr helm chart`
- Might be good to document the var for redis?

**Review by mjnagel — APPROVED** [review]

**Pull request merged**